public/app/features/inspector/utils/download.ts

```diff
@@ -0,0 +1,79 @@
+import saveAs from 'file-saver';
+
+import {
+  CSVConfig,
+  DataFrame,
+  DataTransformerID,
+  dateTimeFormat,
+  dateTimeFormatISO,
+  LogsModel,
+  toCSV,
+} from '@grafana/data';
+
+/**
+ * Downloads a DataFrame as a TXT file.
+ *
+ * @param {(Pick<LogsModel, 'meta' | 'rows'>)} logsModel
+ * @param {string} title
+ */
+export function downloadLogsModelAsTxt(logsModel: Pick<LogsModel, 'meta' | 'rows'>, title: string) {
+  let textToDownload = '';
+
+  logsModel.meta?.forEach((metaItem) => {
+    const string = `${metaItem.label}: ${JSON.stringify(metaItem.value)}\n`;
+    textToDownload = textToDownload + string;
+  });
+  textToDownload = textToDownload + '\n\n';
+
+  logsModel.rows.forEach((row) => {
+    const newRow = dateTimeFormatISO(row.timeEpochMs) + '\t' + row.entry + '\n';
+    textToDownload = textToDownload + newRow;
+  });
+
+  const blob = new Blob([textToDownload], {
+    type: 'text/plain;charset=utf-8',
+  });
+
+  const fileName = `${title}-logs-${dateTimeFormat(new Date())}.txt`;
+  saveAs(blob, fileName);
+}
+
+/**
+ * Exports a DataFrame as a CSV file.
+ *
+ * @param {DataFrame} dataFrame
+ * @param {string} title
+ * @param {CSVConfig} [csvConfig]
+ * @param {DataTransformerID} [transformId=DataTransformerID.noop]
+ */
+export function downloadDataFrameAsCsv(
+  dataFrame: DataFrame,
+  title: string,
+  csvConfig?: CSVConfig,
+  transformId: DataTransformerID = DataTransformerID.noop
+) {
+  const dataFrameCsv = toCSV([dataFrame], csvConfig);
+
+  const blob = new Blob([String.fromCharCode(0xfeff), dataFrameCsv], {
+    type: 'text/csv;charset=utf-8',
+  });
+
+  const transformation = transformId !== DataTransformerID.noop ? '-as-' + transformId.toLocaleLowerCase() : '';
+  const fileName = `${title}-data${transformation}-${dateTimeFormat(new Date())}.csv`;
+  saveAs(blob, fileName);
+}
+
+/**
+ * Downloads any object as JSON file.
+ *
+ * @param {*} json
+ * @param {string} title
+ */
+export function downloadAsJson(json: any, title: string) {
```

Comment:
Don't use `any`. This type should not be in our vocabulary!

[`unknown` should work](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAEzgdzAGzgQ2QQQGcApQhACgCsywAuRcAazHTAEpEBvAKEUQCcAplBD8kxAMoB5AHIA6QlH4wwAcxjAAnlRpsA3NwC+3bqgzY8RUhQBEAC0GZsAGkRo4-TMhv7TrCwQkNOScDk5w9Dbunt6GvmZYuIHWYOQA2vaO2DauUR5eNgC6+kA)

Reply:
Thanks Josh! 🙏